app/src/main/java/com/realityexpander/tasky/agenda_feature/data/repositories/agendaRepository/agendaRepositoryImpls/AgendaRepostoryImpl.kt

```diff
@@ -35,24 +35,38 @@ class AgendaRepositoryImpl @Inject constructor(
 
     override fun getAgendaForDayFlow(dateTime: ZonedDateTime): Flow<List<AgendaItem>> {
 
-        return flow {
+        return channelFlow {
             supervisorScope {
-                launch {
+                launch(Dispatchers.IO) {
                     try {
+                        send(eventRepository.getEventsForDay(dateTime)) // send stale local data
+
+                        // Get fresh data
                         val result = agendaApi.getAgenda(dateTime)
 
-                        //eventRepository.clearEventsForDay(dateTime)
+//                        eventRepository.clearEventsForDayLocally(dateTime) // clear local data // todo add this
                         result.events.forEach { event ->
-                            eventRepository.upsertEventLocally(event.toDomain())
+                            // Insert fresh data into db
+                            val result2 =
+                                eventRepository.upsertEventLocally(event.toDomain())
+                            if(result2 is ResultUiText.Error) {
+                                throw IllegalStateException(result2.message.asStrOrNull())
+                            }
+                        }
+
+                        // Emit the current state from DB
+                        eventRepository.getEventsForDayFlow(dateTime).collect { events ->
+                            send(events)
                         }
-                    } catch (e: CancellationException) {
-                        throw e
-                    } catch (e: Exception) {
+                    }
+                    catch (e: CancellationException) {
+                        /* intentionally eat this exception */
```

Comment:
Why?

Reply:
Im still a little confused when to catch `CancellationExceptions`. 

In this particular case figured that ignoring a `CancellationException` would be appropriate, as the calling source doesnt need to know it was cancelled and will be handled by the `supervisorScope`

## FOLLOW UP QUESTION 2

Should the `CancellationException` be allowed to propagate to the `supervisorScope`?

Reply:
You never really want to catch cancellation exceptions (unless you need the info that the coroutine was cancelled). In all other cases, the framework uses them to properly handle cancellation

Reply:
OK, which video link was that? Ive seen most if not all of your coroutine videos, but I may need a refresher.